File: Documentation/4.0/Samples/nodejs/clientApi/session/querying/documentQuery/whatIsDocumentQuery.js
import { DocumentStore, AbstractIndexCreationTask } from "ravendb";

let documentType, indexName, collectionName, isMapReduce;

const documentStore = new DocumentStore();
const session = documentStore.openSession();

class Employee {
}

class MyCustomIndex extends AbstractIndexCreationTask { }

function docQueryIface() {
    //region document_query_1
    session.advanced.documentQuery(documentType);

    session.advanced.documentQuery({
        indexName,
        collectionName,
        isMapReduce,
        documentType
    });
    //endregion
}

async function docQueryExamples() {
    {
        //region document_query_2
        // load all entities from 'Employees' collection
        const employees = await session.advanced
            .documentQuery(Employee)
            .all();

        // same as
        const employees2 = await session.advanced
            .documentQuery({ collection: "Employees" })
            .all();
        //endregion
    }

    {
        //region document_query_3
        // load all entities from 'Employees' collection
        // where firstName equals 'Robert'
        const employees = await session.advanced
            .documentQuery({ collection: "Employees" })
            .whereEquals("FirstName", "Robert")
            .all();
        //endregion
    }

    {
        //region document_query_4
        // load all entities from 'Employees' collection
        // where firstName equals 'Robert'
        // using 'My/Custom/Index'
        const employees = await session.advanced
            .documentQuery({ indexName: "My/Custom/Index" })
            .whereEquals("FirstName", "Robert")
            .all();
        //endregion
    }
}

File: Documentation/4.1/Samples/java/src/test/java/net/ravendb/ClientApi/Session/Querying/DocumentQuery/WhatIsDocumentQuery.java
package net.ravendb.ClientApi.Session.Querying.DocumentQuery;

import net.ravendb.client.documents.DocumentStore;
import net.ravendb.client.documents.IDocumentStore;
import net.ravendb.client.documents.indexes.AbstractIndexCreationTask;
import net.ravendb.client.documents.session.IDocumentQuery;
import net.ravendb.client.documents.session.IDocumentSession;

import java.util.List;

public class WhatIsDocumentQuery {

    private class Employee {
    }

    private class MyCustomIndex extends AbstractIndexCreationTask {
    }

    private interface IFoo {
        //region document_query_1
        <T> IDocumentQuery<T> documentQuery(Class<T> clazz);

        <T> IDocumentQuery<T> documentQuery(Class<T> clazz, String indexName, String collectionName, boolean isMapReduce);
        //endregion
    }

    public WhatIsDocumentQuery() {
        try (IDocumentStore store = new DocumentStore()) {
            try (IDocumentSession session = store.openSession()) {
                //region document_query_2
                // load all entities from 'Employees' collection
                List<Employee> employees = session
                    .advanced()
                    .documentQuery(Employee.class)
                    .toList();
                //endregion
            }

            try (IDocumentSession session = store.openSession()) {
                //region document_query_3
                // load all entities from 'Employees' collection
                // where firstName equals 'Robert'
                List<Employee> employees = session
                    .advanced()
                    .documentQuery(Employee.class)
                    .whereEquals("firstName", "Robert")
                    .toList();
                //endregion
            }

            try (IDocumentSession session = store.openSession()) {
                //region document_query_4
                // load all entities from 'Employees' collection
                // where firstName equals 'Robert'
                // using 'My/Custom/Index'
                List<Employee> employees = session
                    .advanced()
                    .documentQuery(Employee.class, "My/Custom/Index", null, false)
                    .whereEquals("firstName", "Robert")
                    .toList();
                //endregion
            }

            try (IDocumentSession session = store.openSession()) {
                //region document_query_5
                // load all entities from 'Employees' collection
                // where firstName equals 'Robert'
                // using 'My/Custom/Index'
                List<Employee> employees = session
                    .advanced()
                    .documentQuery(Employee.class, MyCustomIndex.class)
                    .whereEquals("firstName", "Robert")
                    .toList();
                //endregion
            }
        }
    }
}
